packages/mikro-orm/src/entities.ts

```diff
@@ -0,0 +1,146 @@
+import { v4 as randomUUID } from "uuid"
+
+import {
+  Property,
+  Unique,
+  PrimaryKey,
+  Entity,
+  Enum,
+  OneToMany,
+  Collection,
+  ManyToOne,
+  Cascade,
+} from "@mikro-orm/core"
+import type { DefaultAccount } from "next-auth"
+import type {
+  AdapterSession,
+  AdapterUser,
+  VerificationToken as AdapterVerificationToken,
+} from "next-auth/adapters"
+import type { ProviderType } from "next-auth/providers"
+
+type RemoveIndex<T> = {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  [K in keyof T as {} extends Record<K, 1> ? never : K]: T[K]
+}
+
+@Entity()
+export class User implements RemoveIndex<AdapterUser> {
+  @PrimaryKey()
+  id: string = randomUUID()
+
+  @Property({ nullable: true })
+  name?: string
+
+  @Property({ nullable: true })
+  @Unique()
+  email?: string
+
+  @Property({ type: "Date", nullable: true })
+  emailVerified: Date | null = null
+
+  @Property({ nullable: true })
+  image?: string
+
+  @OneToMany({
+    entity: () => Session,
+    mappedBy: (session) => session.user,
+    hidden: true,
+    orphanRemoval: true,
+    cascade: [Cascade.ALL],
+  })
+  sessions = new Collection<Session>(this)
+
+  @OneToMany({
+    entity: () => Account,
+    mappedBy: (account) => account.user,
+    hidden: true,
+    orphanRemoval: true,
+    cascade: [Cascade.ALL],
+  })
+  accounts = new Collection<Account>(this)
+}
+
+@Entity()
+export class Session implements AdapterSession {
+  @PrimaryKey({ default: randomUUID() })
```

Comment:
this is wrong, the `default` option in decorator is for database defaults, here you would give it a specific UUID generated on require time

Reply:
I think this is a debugging / try-around leftover when it didn't work with sqlite - good catch.